src/Libraries/Microsoft.Extensions.Compliance.Abstractions/Classification/DataClassificationSet.cs

```diff
@@ -0,0 +1,144 @@
+    {
+        _ = Throw.IfNull(classifications);
+        _classifications.UnionWith(classifications);
+    }
+
+    /// <summary>
+    /// Converts a data classification to a data classification set.
+    /// </summary>
+    /// <param name="classification">The classification to include in the set.</param>
+    public static implicit operator DataClassificationSet(DataClassification classification)
+    {
+        return FromDataClassification(classification);
+    }
+
+    /// <summary>
+    /// Converts a data classification to a data classification set.
+    /// </summary>
+    /// <param name="classification">The classification to include in the set.</param>
+    /// <returns>The resulting data classification set.</returns>
+    public static DataClassificationSet FromDataClassification(DataClassification classification) => new(classification);
+
+    /// <summary>
+    /// Creates a new data classification that combines the current classifications with another set.
+    /// </summary>
+    /// <param name="other">The other set.</param>
+    /// <returns>The resulting set which combines the current instance's classifications and the other ones.</returns>
+    public DataClassificationSet Union(DataClassificationSet other)
+    {
+        _ = Throw.IfNull(other);
+
+        var result = new DataClassificationSet(other._classifications);
+        result._classifications.UnionWith(_classifications);
+
+        return result;
+    }
+
+    /// <summary>
+    /// Gets a hash code for the current object instance.
+    /// </summary>
+    /// <returns>The hash code value.</returns>
+    public override int GetHashCode() => _classifications.GetHashCode();
+
+    /// <summary>
+    /// Compares an object with the current instance to see if they contain the same data classes.
+    /// </summary>
+    /// <param name="obj">The other data classification to compare to.</param>
+    /// <returns><see langword="true"/> if the two sets contain the same classifications.</returns>
+    public override bool Equals(object? obj) => Equals(obj as DataClassificationSet);
+
+    /// <summary>
+    /// Compares an object with the current instance to see if they contain the same data classes.
+    /// </summary>
+    /// <param name="other">The other data classification to compare to.</param>
+    /// <returns><see langword="true"/> if the two sets contain the same classifications.</returns>
+    public bool Equals(DataClassificationSet? other)
+    {
+        if (other == null)
+        {
+            return false;
+        }
+
+        return _classifications.SetEquals(other._classifications);
+    }
+
+    /// <summary>
+    /// Returns a string representation of this object.
+    /// </summary>
+    /// <returns>The string representation of this object.</returns>
+    public override string ToString()
+    {
+        var sb = PoolFactory.SharedStringBuilderPool.Get();
+
+        var a = _classifications.ToArray();
+        Array.Sort(a, (x, y) =>
```

Comment:
Is sorting strongly needed here?

Reply:
I think so, otherwise ToString would produce non-deterministic results.

Reply:
There's no mention or requirement in the guidelines for `ToString()` to be deterministic or not: https://learn.microsoft.com/en-us/dotnet/api/system.object.tostring?view=net-7.0#notes-to-inheritors

IMO the `ToString()` shouldn't allocate and do sorting every time. Can we maybe cache the resulted string once it's constructed and return it in subsequent calls? There's no way to modify the `_classifications`, right?

Reply:
We could cache the string, but I'm not sure it's worth it. I expect this ToString to be effectively never be used, except via the debugger. I could just remove ToString completely...

Reply:
Do note that many ToString calls allocate in order to form a string. And note that the vast vast majority of cases of DataClassificationSet will have a single element in it, so sorting will be a noop.

Reply:
Okay, thanks